source/SpinalHDL/Structuring/parametrization.rst

```diff
@@ -15,7 +15,41 @@ There is multiple aspect to parametrization :
 Elaboration time parameters
 ------------------------------------------
 
-You can use the whole scala syntax to provide elaboration time parameters. 
+You can use the whole Scala syntax to provide elaboration time parameters.
+
+The whole syntax means you have the entire power and feature set of the
+Scala language at your disposal to solve parametrization requirements for
+your project at the level of complexity you choose.
+
+SpinalHDL does not place any opinionated restrictions on how to achieve
+your parametrization goals.  As such there are many Scala design patterns
+and a few SpinalHDL helpers that can be used to manage parameters that
+are suited to different parameter management scenarios.
+
+Here are some examples and ideas of the possibilities:
+
+ * Hardwired code and constants (not strictly parameter management at all
+   but serves to hilight the most basic mechanism, a code change, not
+   paramater data change)
+ * Constant values provided from a companion object that are static
+   constants in Scala.
+ * Values provided to Scala class constructor, often a ``case class`` that
+   causes Scala to capture those constructor argument values as constants.
+ * Regular Scala flow-control syntax, not limited to but including
+   conditionals, looping, everything.
+ * Config class pattern (examples exist in library items such as
+   UartCtrlConfig, SpiMasterCtrlConfig)
```

Comment:
we could link this to e.g. https://spinalhdl.github.io/SpinalDoc-RTD/master/SpinalHDL/Examples/Intermediates%20ones/uart.html#controller-construction-parameters